install.py

```diff
@@ -17,6 +17,8 @@
 from distutils.spawn import find_executable
 from distutils.sysconfig import get_python_lib
 
+import packaging.version
```

Comment:
What is `packaging`?

```
$ grep -r packaging *
install.py:import packaging.version
install.py:                most_recent_version = packaging.version.parse(
install.py:                candidate_version = packaging.version.parse(
install.py:                most_recent_release = packaging.version.parse(
install.py:                candidate_release = packaging.version.parse(
test-requirements.txt:packaging
Binary file tests/fixtures/rpm/rpm-build-libs-4.13.0.1-2.fc25.x86_64.rpm matches
Binary file tests/fixtures/rpm/rpm-build-libs-4.13.0.1-2.fc25.i686.rpm matches
Binary file tests/fixtures/rpm/rpm-devel-4.13.0.1-2.fc25.i686.rpm matches
Binary file tests/fixtures/rpm/rpm-devel-4.13.0.1-2.fc25.x86_64.rpm matches
tox-requirements.txt:packaging
```

Reply:
This module: https://github.com/pypa/packaging. It provides a convenient version parser that I'm using to get the downloaded rpms from zypper (zypper doesn't have a "download it into this directory" option, it can only download the rpm into the default cache directory, from which I am extracting the rpm).

Reply:
For example you can run `install.py` directly without using `pip` command like this.
But this command outputs the error.
Sorry I did not include this use case in the unit test.

```
$ python3 -m venv ./venv

$ source venv/bin/activate

(venv) $ python -c "$(curl -fsSL https://raw.githubusercontent.com/dcermak/rpm-py-installer/master/install.py)"
Traceback (most recent call last):
  File "<string>", line 20, in <module>
ModuleNotFoundError: No module named 'packaging'
(venv) $ 
```

Reply:
Darn, I haven't thought about that when adding the dependency on packaging.

Reply:
That we can not use pip packages as a dependency in `install.py` is a kind of current challenge for `rpm-py-installer` :<

> https://github.com/junaruga/rpm-py-installer/blob/master/docs/developers_guide.md
> The repository has below logic files. The every logic is included in the install.py. The reason is to run it directory from curl command without pip. I think that is useful to debug before actual release. But it makes the source code hard to read. If we can compress the logic from .py files to one py file including zipped binary code like get-pip.py, we can split the install.py to several py files.

Is it possible to do "get the downloaded rpms from zypper" without using `packaging`? `install.py` has `Cmd.curl_remote_name` class method to download a file from web like curl command.

Reply:
> Is it possible to do "get the downloaded rpms from zypper" without using `packaging`?

Of course, I can implement the version parsing and comparison myself. It's just more work.

Reply:
I went a little spelunking into the code base and found that you actually already implemented version parsing (albeit simpler but good enough for RPM version & releases). Let's see whether Travis is happy too, as it works on my machine™

Reply:
Oh that is `Utils.version_str2tuple`. I forget the existence. Travis is happy too!

Reply:
I tried to run above command to install rpm-py-installer directly without pip.

```
$ python --version
Python 3.7.4

$ source venv/bin/activate
(venv) $ python -c "$(curl -fsSL https://raw.githubusercontent.com/dcermak/rpm-py-installer/master/install.py)"
[INFO] Installing...
[INFO] Created working directory '/tmp/tmpez5bt5dr-rpm-py-installer'
[INFO] Downloading archive. 'https://github.com/rpm-software-management/rpm/archive/rpm-4.14.2.1-release.tar.gz'.
[INFO] Removed working directory '/tmp/tmpez5bt5dr-rpm-py-installer'
[INFO] Done successfully.
(venv) $ Traceback (most recent call last):
```

Then I got a unknown message "Traceback (most recent call last):".
Do you see it on your machine?

```
(venv) $ pip list
Package    Version 
---------- --------
pip        19.0.3  
rpm        4.14.2.1
setuptools 40.8.0  
You are using pip version 19.0.3, however version 19.2.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

Reply:
The message "Traceback (most recent call last):" is not outputted for `install.py` on this repository's latest master branch.

```
(venv_current_master) $ python -c "$(curl -fsSL https://raw.githubusercontent.com/junaruga/rpm-py-installer/master/install.py)"
[INFO] Installing...
[INFO] Created working directory '/tmp/tmp4_fsh_5g-rpm-py-installer'
[INFO] Downloading archive. 'https://github.com/rpm-software-management/rpm/archive/rpm-4.14.2.1-release.tar.gz'.
[INFO] Removed working directory '/tmp/tmp4_fsh_5g-rpm-py-installer'
[INFO] Done successfully.
(venv_current_master) $ 
```

Reply:
Hm, unfortunately I cannot reproduce this additional `Traceback (most recent call last):` output. In which environment did you run it? I've tried a fresh Tumbleweed, Leap 15.1 and Fedora 30 Vagrant box.

Reply:
My environment where I faced the issue is, my local machine. Fedora 30.
The python is like this.
But surprisingly when I tried it again today, it does not show the `Traceback ...` message. It is no problem. :)

```
$ which python3
/usr/bin/python3

$ rpm -qf /usr/bin/python3
python3-3.7.4-1.fc30.x86_64

$ /usr/bin/python3 --version
Python 3.7.4

$ python3 -m venv ./ven

$ source venv/bin/activate

(venv) $ pip list
Package    Version
---------- -------
pip        19.0.3 
setuptools 40.8.0 
You are using pip version 19.0.3, however version 19.2.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

(venv) $ python -c "$(curl -fsSL https://raw.githubusercontent.com/dcermak/rpm-py-installer/master/install.py)"
[INFO] Installing...
[INFO] Created working directory '/tmp/tmpe1ciftxx-rpm-py-installer'
[INFO] Downloading archive. 'https://github.com/rpm-software-management/rpm/archive/rpm-4.14.2.1-release.tar.gz'.
[INFO] Removed working directory '/tmp/tmpe1ciftxx-rpm-py-installer'
[INFO] Done successfully.

(venv) $ pip list
Package    Version 
---------- --------
pip        19.0.3  
rpm        4.14.2.1
setuptools 40.8.0  
You are using pip version 19.0.3, however version 19.2.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

(venv) $ python -c "import rpm; print(rpm.__version__)"
4.14.2.1
```